File: subwatch-api/Models/Entities/PriceHistory.cs
namespace SubwatchApi.Models.Entities
{
    public class PriceHistory
    {
        public int Id { get; set; }
        public decimal Price { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        public int SubscriptionId { get; set; }
        public Subscription Subscription { get; set; } = null!;
    }
}

File: subwatch-frontend/src/api/price-history.ts
import { mainApi } from "../lib/axios-instance";
import type { PriceHistoryResponse, CreatePriceHistoryRequest } from "./types";

export async function createPricehistory({ price, startDate, endDate, subscriptionId }: CreatePriceHistoryRequest) {
    const data = {
        price,
        startDate,
        endDate,
        subscriptionId
    }

    const response = await mainApi.post<PriceHistoryResponse>("/pricehistories", data)
    return response.data
}

export async function getPricehistory(id: number) {
    const response = await mainApi.get<PriceHistoryResponse>(`/pricehistory/${id}`)
    return response.data
}

export async function getPriceHistories() {
    const response = await mainApi.get<PriceHistoryResponse[]>("/pricehistories")
    return response.data
}

export async function deletePricehistory(id: number) {
    await mainApi.delete(`/pricehistory/${id}`)
}